src/bepasty/utils/date_funcs.py

```diff
@@ -1,17 +1,26 @@
 import time
 from flask import current_app
+from werkzeug.exceptions import BadRequest
 
 from ..constants import FOREVER, TIMESTAMP_MAX_LIFE
 
 
 def get_maxlife(data, underscore):
     unit_key = 'maxlife_unit' if underscore else 'maxlife-unit'
-    unit_default = 'MONTH'
+    unit_default = 'MONTHS'
     unit = data.get(unit_key, unit_default).upper()
     value_key = 'maxlife_value' if underscore else 'maxlife-value'
     value_default = '1'
-    value = int(data.get(value_key, value_default))
-    return time_unit_to_sec(value, unit)
+    try:
+        value = int(data.get(value_key, value_default))
+    except (ValueError, TypeError):
+        field = value_key if underscore else 'Maxlife-Value'
+        raise BadRequest(description='{} header is incorrect'.format(field))
```

Comment:
remove 17 and just use value_key instead of field?

Reply:
other header fields uses camel case. if you don't care inconsistency, value_key would be fine.

Reply:
i think 17 is just inconsistent.

we access via value_key, always.

so the error message should not use value_key for one case and something hardcoded for the other.

Reply:
users of get_maxlife() are, one is request.form and one is request.headers.

And questionable one is request.headers, because it is case insensitive dict, and the header is usually using camel case. If we don't care this camel case in request.headers,

for example,

```python
        if not request.headers.get("Content-Range"):
            return 'Content-Range not specified', 400
```

Content-Range error uses camel case. So actually I myself don't care lower or camel though, I wanted to confirm you also don't care this.

Reply:
don't care, we'll just use the key (which we really use to access the dict) also for the message.

Reply:
the idea to remove "field = ..." stuff, how about the following change? using camel case for *_key

```python
def get_maxlife(data, underscore):
    unit_key = 'maxlife_unit' if underscore else 'Maxlife-Unit'
    unit_default = 'MONTHS'
    unit = data.get(unit_key, unit_default).upper()
    value_key = 'maxlife_value' if underscore else 'Maxlife-Value'
    value_default = '1'
    try:
        value = int(data.get(value_key, value_default))
    except (ValueError, TypeError):
        raise BadRequest(description='{} header is incorrect'.format(value_key))
    try:
        return time_unit_to_sec(value, unit)
    except KeyError:
        raise BadRequest(description='{} header is incorrect'.format(unit_key))
```

Reply:
i guess within the current commits in this PR, you have the least possible changes, so keep it as is.

in general, i do not like huge pull requests and i especially do not like even more huge pull requests that additionally "fix" (change) stuff that was **not** broken.

the work you do within this PR is great and very welcome (esp. that you added a lot of tests), but let's concentrate on real fixes here.

Reply:
I agree, but at same time disagree. making clean history is good and very preferred, however, it takes more time and cost. so we would need a balance as usual. and IMO, at least, making forward and better result should be more important than cleaner history.

anyway, i already reverted those changes.